client/js/lounge.js

```diff
@@ -111,6 +111,64 @@ $(function() {
 		index: 1
 	};
 
+	const colorCodeMap = [
+		[0, "White"],
+		[1, "Black"],
+		[2, "Blue"],
+		[3, "Green"],
+		[4, "Red"],
+		[5, "Brown"],
+		[6, "Magenta"],
+		[7, "Orange"],
+		[8, "Yellow"],
+		[9, "Light Green"],
+		[10, "Cyan"],
+		[11, "Light Cyan"],
+		[12, "Light Blue"],
+		[13, "Pink"],
+		[14, "Grey"],
+		[15, "Light Grey"],
+	];
+
+	const foregroundColorStrategy = {
+		id: "foreground-colors",
+		match: /\B\x03(\d{0,2}|[A-Za-z ]{0,10})$/,
+		search(term, callback) {
+			term = term.toLowerCase();
+			const matchingColorCodes = colorCodeMap
+				.filter(i => i[0].toString().startsWith(term) || i[1].toLowerCase().startsWith(term));
+
+			callback(matchingColorCodes);
+		},
+		template(value) {
+			return `<span class="irc-fg${value[0]}">${value[1]}</span>`;
+		},
+		replace(value) {
+			return "\x03" + value[0];
+		},
+		index: 1
+	};
+
+	const backgroundColorStrategy = {
+		id: "background-colors",
+		match: /\B\x03(\d{1,2}),(\d{0,2}|[A-Za-z ]{0,10})$/,
+		search(term, callback, match) {
+			term = term.toLowerCase();
+			const matchingColorCodes = colorCodeMap
+				.filter(i => i[0].toString().startsWith(term) || i[1].toLowerCase().startsWith(term))
+				.map(pair => pair.concat(match[1])); // Needed to pass fg color to `template`...
+
+			callback(matchingColorCodes);
+		},
+		template(value) {
+			return `<span class="irc-fg${value[2]} irc-bg${value[0]}">${value[1]}</span>`;
+		},
+		replace(value) {
+			return "\x03$1," + value[0];
```

Comment:
I sure can, but do we need to, knowing that it works fine as is? Asking because I don't know the IRC details, you probably know a reason I don't.